File: Visual Basic/ImportExcel.vb
'Import excel references.
'References > COM > Microsoft Excel 16.0 Object Library
Imports Excel = Microsoft.Office.Interop.Excel

Private Function ImportExcel(ByVal file As String) As DataTable
    Dim dt As New DataTable
    Dim colName As String

    Try
        Dim xlApp As Excel.Application = New Excel.Application()
        Dim xlWorkbook As Excel.Workbook = xlApp.Workbooks.Open(file)
        Dim xlWorkSheet As Excel._Worksheet = xlWorkbook.Sheets(1)          'active sheet for data
        Dim xlRange As Excel.Range = xlWorkSheet.UsedRange
        Dim Values As Object(,) = xlRange.Value
        Dim rowCount As Integer = xlRange.Rows.Count
        Dim colCount As Integer = xlRange.Columns.Count

        'COLUMNS
        For i As Integer = 1 To colCount
            'start index Columns is row 1
            colName = Convert.ToString(Values(1, i))

            If colName IsNot Nothing OrElse colName <> "" Then
                dt.Columns.Add(colName)
            End If
        Next

        'ROWS
        For r As Integer = 2 To rowCount
            'start index Data is row 2
            Dim row As DataRow = dt.NewRow()

            For c As Integer = 1 To colCount
                Dim field As String = Convert.ToString(Values(r, c))
                row(c - 1) = field
            Next

            dt.Rows.Add(row)
        Next

        'cleanup
        GC.Collect()
        GC.WaitForPendingFinalizers()
        'release COM Objects to kill Excel process from running in the background
        Marshal.ReleaseComObject(xlRange)
        Marshal.ReleaseComObject(xlWorkSheet)
        'close and release
        xlWorkbook.Close()
        Marshal.ReleaseComObject(xlWorkbook)
        'quit and release
        xlApp.Quit()
        Marshal.ReleaseComObject(xlApp)

    Catch ex As Exception
        Throw
    End Try

    Return dt
End Function

File: _ClassesToUse/ImportExcel.cs
using System;
using System.Data;
using Excel = Microsoft.Office.Interop.Excel;

public static class ImportExcel 
{
    //import data from excel into DataTabel

    public static DataTable ImportExcel(string filePathExcel)
    {
        DataTable dt = new DataTable();
        int sheet    = 0;   //number Excelsheet
        int colIndex = 0;   //index Column start
        int rowIndex = 0;   //index Row start

        try 
        {
            Excel.Application   xlApp           = new Excel.Application();
            Excel.Workbook      xlWorkbook      = xlApp.Workbook.Open(filePathExcel);
            Excel._Worksheet    xlWorksheet     = xlWorkbook.Sheets[sheet];
            Excel.Range         xlRange         = xlWorksheet.UsedRange;

            int rowCount = xlRange.Rows.Count;
            int colCount = xlRange.Columns.Count;
            object[,] values = xlRange.Value;

            string col;
            string cell;
            
            //  [COLUMNS]
            for (int c = colIndex - 1; c <= colCount; c++)
            {
                col = values[rowIndex - 1, c].ToString();
                if (col != null || col != "")
                {
                    col = col.Trim();
                    dt.columns.Add(col);
                }
            }

            //  [ROWS]
            for (int r = rowIndex - 1; r <= rowCount; r++)
            {
                DataRow row = dt.NewRow();
                for (int c = colIndex - 1; c<= colCount; c++)
                {
                    if (values[r, c] != null) 
                    {
                        cell = values[r, c].ToString();
                    }
                    else 
                    {
                        cell = "";
                    }
                    row[c - 1] = cell;
                }
                dt.Rows.Add(row);
            }

            // Cleanup
            GC.Collect();
            GC.WaitForPendingFinalizers();
            // Release COM Objects to kill excel process from running in the background
            Marshal.ReleaseComObject(xlRange);
            Marshal.ReleaseComObject(xlWorksheet);
            // Close and release
            xlWorkbook.Close();
            Marshal.ReleaseComObject(xlWorkbook);
            // Quit and release
            xlApp.Quit();
            Marshal.ReleaseComObject(xlApp);
        }
        catch(Exception ex)
        {
            throw;
        }

        return dt;
    }
}